### PR TITLE
JV-1436: added description for max_bars_back error

### PR DESCRIPTION
**Pull request by mtrofimchuk**
Added description for max_bars_back error

### CODE REVIEW

File: source/appendix/Pine_compilation_errors.rst [hunk 1]
```diff
@@ -160,3 +160,31 @@ can be сonverted into::
...
     var3 = expr1 + expr2
 
+
+Pine miscalculated length of series, required length is N. Try using max_bars_back=N in study() function
+---------------------------------------------------------------------------------------------------------
+
+The error appears in case Pine wrongly autodetected required max length of
+series used in the script. When the script is trying to get a value N bars
+ago, calculation fails. Most likely it happens when a script has an 
+``if``-condition with a branch referencing too far back in history. And if
```
Comment (kuznet1) [inline source/appendix/Pine_compilation_errors.rst]: add "or ternary"

File: source/appendix/Pine_compilation_errors.rst [hunk 2]
```diff
@@ -160,3 +160,31 @@ can be сonverted into::
...
+
+    //@version=4
+    study("Requires max_bars_back")
+    test = 0.0
+    if bar_index > 1000
+        test := vwma(close, 20)
+    plot(test)
+
+In order to help Pine with detection, you should use add ``max_bars_back``
+param in ``study`` function::
```
Comment (kuznet1) [inline source/appendix/Pine_compilation_errors.rst]: or, you can take out problematic expression from condition like this:
```
    //@version=4
    study("Requires max_bars_back")
    test = 0.0
    vwma20 = vwma(close, 20)
    if bar_index > 1000
        test := vwma20
    plot(test)
```

File: source/appendix/Pine_compilation_errors.rst [hunk 3]
```diff
@@ -164,12 +164,12 @@ can be сonverted into::
...
+The error appears in case Pine wrongly autodetected required max length of 
+series used in the script. When the script is trying to get a value N bars 
 ago, calculation fails. Most likely it happens when a script has an 
-``if``-condition with a branch referencing too far back in history. And if
-a flow of execution seldom attends this branch, so it difficult for Pine to
-detect the length.
+``if``-condition (or ternary operator ``?``) with a branch referencing too 
+far back in history. And if a flow of execution seldom attends this branch, 
+so it difficult for Pine to detect the length.
 An example of the script with the problem::
```
Comment (LucFF) [inline source/appendix/Pine_compilation_errors.rst]: The error appears in cases where Pine wrongly autodetects the required maximum length of series used in a script. This happens when a script's flow of execution does not allow Pine to inspect the use of series in branches of conditional statements (``if``, ``iff`` or ``?``), and Pine cannot automatically detect how far back the series is referenced. Here is an example of a script causing this problem::

File: source/appendix/Pine_compilation_errors.rst [hunk 4]
```diff
@@ -179,7 +179,7 @@ An example of the script with the problem::
         test := vwma(close, 20)
     plot(test)
 
-In order to help Pine with detection, you should use add ``max_bars_back``
+In order to help Pine with detection, you should add ``max_bars_back`` 
 param in ``study`` function::
```
Comment (LucFF) [inline source/appendix/Pine_compilation_errors.rst]: Why do we recommend using the ``study()`` parameter rather than the v4 ``max_bars_back()`` function?

Comment (kuznet1) [reply source/appendix/Pine_compilation_errors.rst]: Unfortunately, `max_bars_back()` func does not help with built-in functions :cry:

File: source/appendix/Pine_compilation_errors.rst (hunk 3 above)
Comment (kuznet1) [reply source/appendix/Pine_compilation_errors.rst]: :+1:

File: source/appendix/Pine_compilation_errors.rst [hunk 5]
```diff
@@ -160,3 +160,41 @@ can be сonverted into::
...
+
+    //@version=4
+    study("Requires max_bars_back")
+    test = 0.0
+    if bar_index > 1000
+        test := vwma(close, 20)
+    plot(test)
+
+In order to help Pine with detection, you should add ``max_bars_back`` 
+param in ``study`` / ``stragety`` function::
```
Comment (anastasianaumenko) [inline source/appendix/Pine_compilation_errors.rst]: опечатка в слове ``stragety``

**Pull request merged**